### PR TITLE
Passing Data Between Components lesson: Refeactor `DefaultProps` Section to Use Default Parameters

### LINKED ISSUE #27882: React: Update lesson to remove `defaultProps`
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### Describe your suggestion

React has announced that support for `defaultProps` in function components will be deprecated in a future major release. We should update the lesson[ Passing Data Between Components](https://www.theodinproject.com/lessons/node-path-react-new-passing-data-between-components) to reflect the upcoming deprecation of `defaultProps`.

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/node-path-react-new-passing-data-between-components

### (Optional) Discord Name

headlessNode

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by headlessNode**
## Because
This pull request refactors the `Default Props` section of the lesson [Passing Data Between Components](https://www.theodinproject.com/lessons/node-path-react-new-passing-data-between-components) to prioritize the use of `default parameters` in function components. This approach aligns with modern JavaScript practices and React's preferred method for setting default prop values.

## This PR
- Refactor the `DefaultProps` section to prioritize default parameters in function components.
- Provides an example demonstrating using default parameters to set default prop values.
- Includes a brief explanation and example of defaultProps for awareness, especially in legacy codebases and class components.

## Issue
Closes #27882

## Additional Information
None.

## Pull Request Requirements
-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Comment by MaoShizhong** [comment]
Feel free to fix the anchor tags as part of this PR. The our `fix:lesson` script should cover them.

**Comment by headlessNode** [comment]
@MaoShizhong done

File: react/getting_started_with_react/passing_data_between_components.md [hunk 1]
```diff
@@ -123,7 +123,32 @@ export default function App() {
...
+      <div>
+        <Button />
+        <Button text="Don't Click Me!" color="red" />
+        <Button fontSize={20} />
+      </div>
+    );
+  }
+```
+
+As you can see, we now only need to supply prop values to `Button` when rendering within `App` if they differ from the default values defined in the function parameters. This approach is simpler and aligns with JavaScript's default parameter syntax.
```
Comment (MaoShizhong) [inline react/getting_started_with_react/passing_data_between_components.md:149]: We won't need the last sentence. It's not React syntax that's similar to JS's default param syntax, it *is* JS's default param syntax. Just plain old JS.

```suggestion
As you can see, we now only need to supply prop values to `Button` when rendering within `App` if they differ from the default values defined in the function parameters.
```

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Just the one bit below but otherwise looks great!

File: react/getting_started_with_react/passing_data_between_components.md (hunk 1 above)
Comment (headlessNode) [reply react/getting_started_with_react/passing_data_between_components.md]: Done

**Review by MaoShizhong — APPROVED** [review]
:rocket:

**Pull request merged**